src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -481,6 +481,13 @@ impl<S: MetaStore> HummockManager<S> {
             .id_gen_manager()
             .generate::<{ IdCategory::CompactionGroup }>()
             .await?;
+        let new_sst_start_id = self
+            .env
+            .id_gen_manager()
+            .generate_interval::<{ IdCategory::HummockSstableId }>(
+                parent_group.count_ssts() as u64 * 2,
```

Comment:
> Why do we need to pre-generate `parent_group.count_ssts() as u64 * 2` sst ids? The id generator already pre-allocates ids in its implementation so we can just use it to generate one id at a time when needed.

Each call to id generator requires a lock and it's slow.

Reply:
> > Each call to id generator requires a lock and it's slow.
> 
> I see. Generally the id generator usage in our codebase will use the generated ids unless there are node/network failures but the usage here will create gaps for sure on every group split.
> 
> Can we generate and assign sst ids after we know how many new ssts are generated? If not possible, can we calculate a rough number of affected ssts beforehand so that we don't explicitly create gaps?

ok